plugin/trino-hive/pom.xml

```diff
@@ -251,6 +256,12 @@
         </dependency>
 
         <!-- used by tests but also needed transitively -->
+        <dependency>
+            <groupId>io.trino</groupId>
+            <artifactId>trino-parser</artifactId>
+            <scope>runtime</scope>
```

Comment:
```
[WARNING] Used undeclared dependencies found:
[WARNING]    io.trino:trino-parser:jar:372-SNAPSHOT:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  25.802 s
[INFO] Finished at: 2022-03-02T06:49:48+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:analyze-only (default) on project trino-hive: Dependency problems found -> [Help 1]
```

Reply:
It was test scope, so not part of production build. Per https://github.com/trinodb/trino/pull/11159#discussion_r816733470 it should remain as not part of build, because we cannot guarantee API compatibility of the trino-parser between versions.